tests/providers/aws/services/sns/sns_topics_not_publicly_accessible/sns_topics_not_publicly_accessible_test.py

```diff
@@ -141,10 +141,10 @@ def test_topic_public_with_condition(self):
             check = sns_topics_not_publicly_accessible()
             result = check.execute()
             assert len(result) == 1
-            assert result[0].status == "PASS"
+            assert result[0].status == "FAIL"
             assert (
                 result[0].status_extended
-                == f"SNS topic {topic_name} is publicly accesible but has a Condition that could filter it"
+                == f"SNS topic {topic_name} is public because its policy allows public access"
```

Comment:
Include a policy with a condition that allows for public access and another one that restricts it.